flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/PipelinedRegionSchedulingStrategy.java

```diff
@@ -53,10 +53,20 @@ public class PipelinedRegionSchedulingStrategy implements SchedulingStrategy {
     private final Map<SchedulingPipelinedRegion, List<ExecutionVertexID>> regionVerticesSorted =
             new IdentityHashMap<>();
 
+    /** All downstream regions of one schedulingPipelinedRegion. */
+    private final Map<SchedulingPipelinedRegion, Set<SchedulingPipelinedRegion>>
+            downstreamRegionsOfRegion = new IdentityHashMap<>();
```

Comment:
It should be good enough to store a mapping from the upstream region to the downstream consumed partition group, as we already have another map helps mapping the groups to downstream regions.

Reply:
Changed it.